src/reference_implementations/prompting_vector_llms/activation_fine_tuning/train_on_activations.ipynb

```diff
@@ -258,26 +285,72 @@
    "source": [
-    "train_dataset = ActivationDataset(os.path.join(activations_path, \"train_activations_with_prompts_demo.pkl\"))\n",
-    "test_dataset = ActivationDataset(os.path.join(activations_path, \"test_activations_with_prompts_demo.pkl\"))"
+    "train_dataset = ActivationDataset(\n",
+    "    os.path.join(activations_path, f\"train_activations_with_prompts_demo_{layer_number_to_load}.pkl\")\n",
+    ")\n",
+    "test_dataset = ActivationDataset(\n",
+    "    os.path.join(activations_path, f\"test_activations_with_prompts_demo_{layer_number_to_load}.pkl\")\n",
+    ")\n",
+    "train_dataloader = DataLoader(train_dataset, batch_size=128, shuffle=True, collate_fn=batch_last_token)\n",
+    "test_dataloader = DataLoader(test_dataset, batch_size=128, shuffle=True, collate_fn=batch_last_token)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now write a relatively simple script to train and evaluate our model. The hidden dimension is small (128) and the final dimension corresponds to our label space (positive, negative).\n",
+    "\n",
+    "__NOTE__: LLaMA-2 activations have a hidden dimension of 5120. On the other hand, if you're using the pre-computed activations for OPT-175B, these activations are much larger at 12,288."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 10,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Epoch: 24: 100%|██████████| 25/25 [00:00<00:00, 43.31it/s, Train-Loss=0.0753, Test-Accuracy=0.887]\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "tensor(0.8867)"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
    "source": [
-    "train_dataloader = DataLoader(train_dataset, batch_size=128, shuffle=True, collate_fn=batch_last_token)\n",
-    "test_dataloader = DataLoader(test_dataset, batch_size=128, shuffle=True, collate_fn=batch_last_token)"
+    "model = MLP({\"embedding_dim\": 5120, \"hidden_dim\": 128, \"label_dim\": 2})\n",
+    "device = \"cuda\" if cuda.is_available() else \"cpu\"\n",
+    "train_and_evaluate_model(model, train_dataloader, test_dataloader, device)"
    ]
   },
   {
    "attachments": {},
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We can now write a relatively simple script to train and evaluate our model. The model has an embedding dimension of 12,288 corresponding to the size of the intermediate activations for OPT-175B. The hidden dimension is small (128) and the final dimension corresponds to our label space (positive, negative)."
+    "This is quite an amazing result. Simply by including a few-shot prompt when computing the activations, we have __significantly__ increased the sampling efficiency of training this small classifier and induced an immense jump in performance."
```

Comment:
I know there is definitely an improvement in performance, but just wondering if we want to characterize it as an immense jump (0.863 -> 0.8867)... 😂 ?

Reply:
Very good point. OPT gets an immense jump in accuracy (hence the wording left over from the previous setup). This is much less substantial. I just noted that it was a "measurable" jump in accuracy.